File: python/problem003.py
"""Finds the largest prime factor of 600851475143"""
def largest_prime(max):
    x = 1
    largest = 0
    while (x < max):
        if max % x == 0:
            if x % 2 != 0 and x % 3 != 0:
                if is_prime(x) == True and x > largest:
                    largest = x
                    print(largest)
        x += 1
    print(largest)
            
            
"""Checks if a number is prime"""
def is_prime(x):
    if x == 1:
        return True
    else:
        i = 2
        prime = True
        while i < x:
            if x % i == 0:
                prime = False
            i += 1
    return prime


largest_prime(600851475143)

File: rust/problem003.rs
/// Calculate the largest prime factor of n

extern crate time;
use time::PreciseTime;

fn find_largest_prime(n : i64) -> i64 {
    let mut largest_prime : i64 = 0;
    let mut i = 2;
    let mut limit = n;
    while i * i <= limit {
        if limit % i == 0 {
            limit /= i;
            largest_prime = i;
        } else {
            i += 1;
        }
    }

    if limit > largest_prime {
        largest_prime = limit;
    }

    return largest_prime;
}

fn main() {
    let start = PreciseTime::now();
    let largest_prime = find_largest_prime(600_851_475_143);
    let end = PreciseTime::now();
    println!("{}", largest_prime);
    println!("Operation took {} seconds", start.to(end));
}
